static/js/tools/stat_var/dataset_selector.tsx

```diff
@@ -0,0 +1,218 @@
+  const [activeSuggestion, setActiveSuggestion] = useState(0);
+  const [datasets, setDatasets] = useState([]);
+  const [filteredSuggestions, setFilteredSuggestions] = useState(
+    Object.keys(props.sourceMap)
+  );
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [userInput, setUserInput] = useState("");
+
+  function updateDatasets(source: string): void {
+    if (!(source in props.sourceMap)) {
+      return;
+    }
+    const dcid = props.sourceMap[source];
+    axios
+      .get(`/api/browser/propvals/isPartOf/${dcid}`)
+      .then((resp) => {
+        const currentDatasets = [];
+        for (const dataset of resp.data?.values?.in) {
+          currentDatasets.push({
+            dcid: dataset.dcid,
+            name: dataset.name,
+          });
+        }
+        currentDatasets.sort((a, b): number => {
+          return a.name.localeCompare(b.name);
+        });
+        setDatasets(currentDatasets);
+      })
+      .catch(() => {
+        setDatasets([]);
+      });
+  }
+
+  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const input = e.currentTarget.value;
+    const newSuggestions = Object.keys(props.sourceMap)
+      .map((s) => {
+        return {
+          name: s,
+          rank: simplify(s).indexOf(simplify(input)),
+        };
+      })
+      .sort((a, b): number => {
+        return a.rank - b.rank;
+      })
+      .filter((s) => s.rank > -1)
+      .map((s) => s.name)
+      .slice(0, MAX_SUGGESTIONS);
+    setFilteredSuggestions(newSuggestions);
+    setUserInput(e.currentTarget.value);
+    setShowSuggestions(true);
+  };
+
+  const handleOnClick = (e: React.MouseEvent<HTMLLIElement>) => {
+    setActiveSuggestion(0);
+    setFilteredSuggestions([]);
+    setShowSuggestions(false);
+    setUserInput(e.currentTarget.innerText);
+    updateDatasets(e.currentTarget.innerText);
+    props.filterStatVars([
+      {
+        dcid: props.sourceMap[e.currentTarget.innerText],
+        name: e.currentTarget.innerText,
+      },
+    ]);
+  };
+
+  const handleOnKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === "Enter") {
+      if (filteredSuggestions.length === 0) {
+        return;
+      }
+      setActiveSuggestion(0);
+      setShowSuggestions(false);
+      if (userInput === "") {
+        setDatasets([]);
+        props.filterStatVars([]);
+      } else {
+        setUserInput(filteredSuggestions[activeSuggestion]);
+        updateDatasets(filteredSuggestions[activeSuggestion]);
+        props.filterStatVars([
+          {
+            dcid: props.sourceMap[filteredSuggestions[activeSuggestion]],
+            name: filteredSuggestions[activeSuggestion],
+          },
+        ]);
+      }
+    } else if (e.key === "ArrowUp") {
+      if (activeSuggestion === 0) {
+        return;
+      }
+      setActiveSuggestion(activeSuggestion - 1);
+    } else if (e.key === "ArrowDown") {
+      if (activeSuggestion - 1 === filteredSuggestions.length) {
```

Comment:
should it be `activeSuggestion === filteredSuggestions.length - 1` which would also be `activeSuggestion + 1 === filteredSuggestions.length`?

Reply:
sorry about that! changed